### DISCUSSION TITLE
Training time of TC2-DDC model

### DISCUSSION THREAD

**Discussion by swarajdalmia** [coqui-ai/TTS#691] [General Q&A]
I'm training TC2-DDC model on 22 hrs of 32khtz data. This is being trained on 8 gpu's, AWS - p2.8x instance using `distribute.py`. However, the training is very slow. Only 8-10 epochs in 24 hrs. Is there something i'm doing wrong ? because i don't think it should be this slow. I'm adding the model configs below. Normally how many epochs can i expect in 24 hrs. 

```
{
    batch_size=16,
    eval_batch_size=16,
    epochs=300,
    gradual_training = [[0, 6, 16], [10, 4, 16], [10000, 3, 16], [20000, 2, 16], [40000, 1, 16]],

    r = 6,
    double_decoder_consistency = True,
    ddc_r = 6,
    max_decoder_steps = 50,
    num_loader_workers=8,
    num_eval_loader_workers=8,
    run_eval=True,
    test_delay_epochs=-1,

    text_cleaner="basic_cleaners",
    phoneme_language="en-us",
    use_phonemes=True,
    min_seq_len = 6,
    max_seq_len = 220,
    enable_eos_bos_chars = True,
    compute_input_seq_cache = True,
    phoneme_cache_path = os.path.join(output_p, "phoneme_cache/"),
    output_path = os.path.join(output_p, "output_path/"),

    print_step=100,
    print_eval=False,
    mixed_precision=False,
    save_step = 10000,
    checkpoint = True
}
```

**Comment by erogol** [comment]
Are you sure the GPUs are working? 

Try the latest dev branch after #685

**Reply by swarajdalmia** [reply]
yes, all the gpu's are working. Yes, i did actually use that version of distribute.py itself.

**Reply by swarajdalmia** [reply]
I'm wondering if there is some issue in parallelism in the code that i can't figure out. Is it possible that the gpu's are all being used but not in parallel ?

**Reply by erogol** [reply]
Have you `Try the latest dev branch after #685`?

**Reply by swarajdalmia** [reply]
i did try but it's not working either.

**Reply by erogol** [reply]
relates to #707
